File: app/src/main/java/com/m1guelgtz/templatecarsapi/Demo/Features/Projects/Domain/Repository/ProjectRepository.kt
package com.m1guelgtz.templatecarsapi.Demo.Features.Projects.Domain.Repository

import com.m1guelgtz.templatecarsapi.Demo.Features.Projects.Domain.Entities.Project
import com.m1guelgtz.templatecarsapi.Demo.Features.Projects.Domain.Entities.ProjectMember

interface ProjectRepository {
    suspend fun listProjects(): List<Project>
    suspend fun createProject(name: String, description: String?): Project
    suspend fun getProject(projectId: String): Project
    suspend fun updateProject(projectId: String, name: String?, description: String?): Project
    suspend fun deleteProject(projectId: String)
    suspend fun listMembers(projectId: String): List<ProjectMember>
    suspend fun addMember(projectId: String, userId: String, role: String)
}

File: app/src/main/java/com/m1guelgtz/templatecarsapi/src_api/domain/repositories/project_repository.py
from abc import ABC, abstractmethod
from uuid import UUID

from src.domain.entities.project import Project
from src.domain.entities.project_member import ProjectMember


class ProjectRepository(ABC):

    @abstractmethod
    async def get_by_id(self, project_id: UUID) -> Project | None:
        ...

    @abstractmethod
    async def get_all(self) -> list[Project]:
        ...

    @abstractmethod
    async def create(self, project: Project) -> Project:
        ...

    @abstractmethod
    async def add_member(self, member: ProjectMember) -> ProjectMember:
        ...

    @abstractmethod
    async def get_members(self, project_id: UUID) -> list[ProjectMember]:
        ...

    @abstractmethod
    async def delete(self, project_id: UUID) -> bool:
        ...
